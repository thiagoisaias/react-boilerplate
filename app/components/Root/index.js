// @flow
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import type { Store } from 'redux';

import App from 'components/App';

type Props = {
  store: Store
};

const Root = (props: Props) => {
  const { store } = props;
  return (
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  );
};

export default Root;
