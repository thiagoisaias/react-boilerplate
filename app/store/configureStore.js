/**
 * Create a Redux Store.
 */

// @flow
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';

/* eslint-disable no-underscore-dangle */
export default function configureStore() {
  const isProductionEnv = process.env.NODE_ENV === 'production';

  // Redux DevTools shouldn't be enabled in production env
  const composeEnhancers = isProductionEnv
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    : compose;

  /* eslint-enable */
  const initialState = {};

  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(thunk))
  );

  if (module.hot) {
    module.hot.accept('./rootReducer', () => {
      store.replaceReducer(rootReducer);
    });
  }

  return store;
}
