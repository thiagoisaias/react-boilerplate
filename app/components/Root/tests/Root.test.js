import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'store/configureStore';
import Root from '../index';

describe('Root', () => {
  const mockProps = {
    store: configureStore()
  };

  const component = shallow(<Root {...mockProps} />);

  it('renders properly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
