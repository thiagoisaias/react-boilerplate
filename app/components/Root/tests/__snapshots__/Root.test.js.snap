// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders properly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Component />
  </BrowserRouter>
</Provider>
`;
